customers/views.py

```diff
@@ -0,0 +1,139 @@
+import customers, json
+from flask import Blueprint, flash, render_template, request
+from app import db
+from customers.models import Customer
+from sqlalchemy import exc
+
+views = Blueprint("views", __name__, template_folder="templates")
+
+
+@views.route("/", methods=["POST"])
+def create():
+    try:
+        # The block of code that is supposed to execute if a new customer is to be placed in the database
+        request_data = request.get_json()
+        userName = request_data["username"]
+        firstName = request_data["first_name"]
+        lastName = request_data["last_name"]
+        email = request_data["email"]
+        dob = request_data["dob"]
+        newCustomer = Customer(
+            username=userName,
+            first_name=firstName,
+            last_name=lastName,
+            email=email,
+            dob=dob,
+        )
+        db.session.add(newCustomer)
+        db.session.commit()
+        return {
+            "id": f"{newCustomer.id}",
+            "username": f"{userName}",
+            "first_name": f"{firstName}",
+            "last_name": f"{lastName}",
+            "email": f"{email}",
+            "dob": f"{dob}",
+            "date_created": f"{newCustomer.date_created}",
+        }
+    except exc.IntegrityError:
+        # Error Code 409, for resource already exists / duplicate resource / Conflict
+        # Error that occurs when a customer already has a place in the database
+        return {"Message": "ERROR, Code: 409, Customer already exists"}
+    except KeyError:
+        # Error that occurs when a user leaves fields empty
+        return {"Message": "ERROR, Code: 422, left required fields empty"}
+
+
+def retrieve(customer_id=-1):
+    try:
+        if customer_id == -1:
+            customer_list = {}
+            for i in range(1, Customer.query.count() + 1):
+                selected_customer = Customer.query.filter_by(id=i).first()
+                customer_list.update({f"Customer {i}": f"{selected_customer.username}"})
+            return json.dumps(customer_list)
+        else:
+            selected_customer = Customer.query.filter_by(id=customer_id).first()
+            returning_data = {
+                "id": f"{selected_customer.id}",
+                "username": f"{selected_customer.username}",
+                "first_name": f"{selected_customer.first_name}",
+                "last_name": f"{selected_customer.last_name}",
+                "email": f"{selected_customer.email}",
+                "dob": f"{selected_customer.dob}",
+                "date_created": f"{selected_customer.date_created}",
+            }
+            return json.dumps(returning_data)
+    except Exception:
+        return "ERROR, Code: 404, invalid id number entered"
+
+
+@views.route("/list", methods=["GET"])
+def list():
+    return retrieve()
```

Comment:
Why did you decide to put the code for this in a nested function?

You should be able to combine `list` and `detail` into a single function named `retrieve` -- maybe you can treat the `customer_id` arg as optional?